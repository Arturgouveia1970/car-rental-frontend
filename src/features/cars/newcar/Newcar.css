.add-car-form {
  margin-left: 370px;
  margin-top: -120px;
}

.text .addcar-btn {
  margin-left: 370px;
}
