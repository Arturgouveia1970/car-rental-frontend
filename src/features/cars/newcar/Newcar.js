import React from 'react';

const NewCar = () => (
  <>
    <h1>Add Car Page</h1>
    <h3>On this page the admin will be able to add new cars to the list</h3>
  </>
);

export default NewCar;
