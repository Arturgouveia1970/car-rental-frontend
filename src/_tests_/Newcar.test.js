import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Newcar from '../components/cars/newcar/Newcar';
import store from '../Redux/configureStore';

describe('New car form tests', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Newcar />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('New car page has a "add" button', () => {
    const register = render(
      <>
        <Provider store={store}>
          <BrowserRouter>
            <Newcar />
          </BrowserRouter>
        </Provider>
      </>,
    );

    expect(register.findByText('Add')).toBeTruthy();
  });
});
