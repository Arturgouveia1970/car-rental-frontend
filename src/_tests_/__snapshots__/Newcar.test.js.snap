// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New car form tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cnt container-fluid vh-100 d-flex flex-column align-items-center"
      >
        <div
          class="textctn"
        >
          <div
            class="txtHeader"
          >
            <h2
              class="zindex fs-1 d-flex text-white align-items-center justify-content-center"
            >
              Add Car
            </h2>
            <p
              class="zindex fs-5 d-flex text-white text-center align-items-center"
            >
              You've got a fancy car from which you want to make some extra bucks? Add it on our website!
            </p>
            <hr
              class="hr"
            />
            <p
              class="zindex mb-5 fs-5 text-white"
            >
              Fill the details of your car.
            </p>
          </div>
          <form
            class="zindex form d-flex justify-content-center mb-0 align-items-end"
          >
            <div
              class="d-flex mb-3"
            >
              <div>
                <label
                  class="form-label text-white ms-3"
                  for="model"
                >
                  Model:
                </label>
                <input
                  aria-label=".form-control-lg example"
                  class="input form-control form-control-lg px-4 mx-2"
                  id="model"
                  placeholder="Model"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <div>
                <label
                  class="form-label text-white ms-3"
                  for="driverName"
                >
                  year:
                </label>
                <input
                  aria-label=".form-control-lg example"
                  class="input form-control form-control-lg px-4 mx-2"
                  id="year"
                  placeholder="Year"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="d-flex mb-3"
            >
              <div>
                <label
                  class="form-label text-white ms-3"
                  for="price"
                >
                  Price per day:
                </label>
                <input
                  aria-label=".form-control-lg example"
                  class="input form-control form-control-lg px-4 mx-2"
                  id="price"
                  min="0"
                  placeholder="Price"
                  required=""
                  type="number"
                  value=""
                />
              </div>
              <div>
                <label
                  class="form-label text-white ms-3"
                  for="picture"
                >
                  Picture:
                </label>
                <input
                  aria-label=".form-control-lg example"
                  class="input form-control form-control-lg px-4 mx-2"
                  id="picture"
                  placeholder="Image Link"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <button
              class="btn btn px-4 ms-4"
              type="submit"
            >
              Add
            </button>
          </form>
          <p
            class="zindex fs-5 fw-semibold"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cnt container-fluid vh-100 d-flex flex-column align-items-center"
    >
      <div
        class="textctn"
      >
        <div
          class="txtHeader"
        >
          <h2
            class="zindex fs-1 d-flex text-white align-items-center justify-content-center"
          >
            Add Car
          </h2>
          <p
            class="zindex fs-5 d-flex text-white text-center align-items-center"
          >
            You've got a fancy car from which you want to make some extra bucks? Add it on our website!
          </p>
          <hr
            class="hr"
          />
          <p
            class="zindex mb-5 fs-5 text-white"
          >
            Fill the details of your car.
          </p>
        </div>
        <form
          class="zindex form d-flex justify-content-center mb-0 align-items-end"
        >
          <div
            class="d-flex mb-3"
          >
            <div>
              <label
                class="form-label text-white ms-3"
                for="model"
              >
                Model:
              </label>
              <input
                aria-label=".form-control-lg example"
                class="input form-control form-control-lg px-4 mx-2"
                id="model"
                placeholder="Model"
                required=""
                type="text"
                value=""
              />
            </div>
            <div>
              <label
                class="form-label text-white ms-3"
                for="driverName"
              >
                year:
              </label>
              <input
                aria-label=".form-control-lg example"
                class="input form-control form-control-lg px-4 mx-2"
                id="year"
                placeholder="Year"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="d-flex mb-3"
          >
            <div>
              <label
                class="form-label text-white ms-3"
                for="price"
              >
                Price per day:
              </label>
              <input
                aria-label=".form-control-lg example"
                class="input form-control form-control-lg px-4 mx-2"
                id="price"
                min="0"
                placeholder="Price"
                required=""
                type="number"
                value=""
              />
            </div>
            <div>
              <label
                class="form-label text-white ms-3"
                for="picture"
              >
                Picture:
              </label>
              <input
                aria-label=".form-control-lg example"
                class="input form-control form-control-lg px-4 mx-2"
                id="picture"
                placeholder="Image Link"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            class="btn btn px-4 ms-4"
            type="submit"
          >
            Add
          </button>
        </form>
        <p
          class="zindex fs-5 fw-semibold"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
